Use JDK 22 on github actions as AGP requires java 17 and above

eyedol opened the pull request:
Fixes [CI issue](https://github.com/sproctor/compose-data-table/actions/runs/12445415216/job/34746704543) where AGP requires Java 17 and above

sproctor commented:
I don't want to use a non-LTS JDK. I changed it to 21. I had removed the JS target from the demo stuff since there's no JS demo. I guess I'll eventually add one, so I merged the JVM and JS implementations for the scrollbar. I've diverged pretty far from this, so I'm going to close it. If there's stuff in it that you think should be merged, create a new PR.

eyedol commented:
> I don't want to use a non-LTS JDK. I changed it to 21. I had removed the JS target from the demo stuff since there's no JS demo. I guess I'll eventually add one, so I merged the JVM and JS implementations for the scrollbar. I've diverged pretty far from this, so I'm going to close it. If there's stuff in it that you think should be merged, create a new PR.

Makes sense. Nah. Nothing in there deserves a merge. Good you closed it.
